### DISCUSSION TITLE
Create detailed documentation and improve support for Xdebug

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/docker-php#408] [Ideas & Feature Requests]
# 👉 Describe the problem
[comment]: <> (What are you trying to solve?)
Some users rely on Xdebug and there are extra steps involved to get it to work.

# 👥 Problem evidence & reach
[comment]: <> (How many people have this problem?)
- https://github.com/serversideup/docker-php/discussions/176

# 🏆 How to solve this problem
[comment]: <> (Describe the feature that you are proposing and how it will solve the problem)

- [ ] Create documentation how to set this up in development
- [ ] Possibly make it easier by providing a standardized INI file that can easily be enabled

# 🥰 Describe the "impact" on users?
[comment]: <> (How will this make people's lives better once it is solved?)

It will dramatically reduce the headache and effort for configuring Xdebug with PHP + Docker.

# 💯 How do we validate the problem is solved?
[comment]: <> (Explain use cases on how we can measure the success of this implementation)

- [ ] Steps are clearly documented how to enable Xdebug
- [ ] Xdebug can easily work in development, but it does not run in production

**Comment by ps-20x** [comment]
Added a PR for spin that would add the logic described by @emilebourquin to the spin laravel basic template: https://github.com/serversideup/spin-template-laravel-basic/pull/19

**Reply by afanjul** [reply]
Why only for laravel? Many users use other frameworks (yii2, symphony, etc.), so the xdebug config should be added to the base "system" for the docker in my opinion.

**Comment by VaguelyOnline** [comment]
Thanks for the great work on this project. 
Did this feature ever get added to the main images? I'm needing xdebug support but I'm unclear from the docs (searching xdebug at https://serversideup.net/open-source/docker-php/docs/getting-started doesn't return anything) whether a flag to enable xdebug was ever added to the docker images? I'm currently going down the route of updating a custom Dockerfile with the following content. However, it seems like having an option and possible a request param to enable xdebug rather than having it always on seems cleaner.

Thanks in advance from a grateful user of your excellent project.

```bash
# Dockerfile.dev
FROM serversideup/php:8.3-fpm-nginx

USER root

# Install Xdebug
RUN install-php-extensions xdebug

# Write Xdebug config
RUN echo "xdebug.mode=debug" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
    && echo "xdebug.start_with_request=trigger" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
    && echo "xdebug.client_host=host.docker.internal" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
    && echo "xdebug.client_port=9003" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini \
    && echo "xdebug.idekey=PHPSTORM" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini

EXPOSE 9003

USER www-data
```
